src/kafunk/Compression.fs

```diff
@@ -52,29 +48,108 @@ module GZip =
     let bz = BinaryZipper(output |> Binary.ofArray)
     MessageSet.Read (messageVer, 0, 0s, output.Length, bz)
 
+[<Compile(Module)>]
+module GZip =
+
+  open System.IO
+  open System.IO.Compression
+
+  let compress =
+    Stream.compress CompressionCodec.GZIP <| fun memStream -> 
+      upcast new GZipStream(memStream, CompressionMode.Compress)
+
+  let decompress =
+    Stream.decompress <| fun memStream -> 
+      upcast new GZipStream(memStream, CompressionMode.Decompress)
+    
+[<Compile(Module)>]
+module Snappy = 
+  
+  open System
+  open Snappy
+
+  module CompressedMessage = 
+    
+    type BinaryZipper with
+      
+      member this.WriteBlock (bytes: ArraySegment<byte>) = 
+        Buffer.BlockCopy(bytes.Array, bytes.Offset, this.Buffer.Array, this.Buffer.Offset, bytes.Count)
+        this.ShiftOffset bytes.Count
+
+      member this.ReadBlock (length: int) =
+        let arr = ArraySegment<byte>(this.Buffer.Array, this.Buffer.Offset, length)
+        this.ShiftOffset length
+        arr
+
+    module private Header =     
+      // Combined size of header in bytes.  
+      let size = 16
+      // Magic string used by snappy-java.
+      let magic = [| byte -126; byte 'S'; byte 'N'; byte 'A'; byte 'P'; byte 'P'; byte 'Y'; byte 0 |]
+      // Current version number taken from snappy-java repo as of 22/05/2017.
+      let currentVer = 1
+      // Minimum compatible version number taken from snappy-java repo as of 22/05/2017.
+      let minimumVer = 1
+
+    let pack (bytes: byte []) =
+      let buf = Array.zeroCreate (Header.size + 4 + bytes.Length)
+      let bz = BinaryZipper(Binary.ofArray buf)
+      
+      // write header compatible with snappy-java
+      bz.WriteBlock(Binary.ofArray Header.magic)
+      bz.WriteInt32(Header.currentVer)
+      bz.WriteInt32(Header.minimumVer)
+
+      // write content
+      // NOTE: this will also write content length in the first 4 bytes, this is expected.
+      bz.WriteBytes(Binary.ofArray bytes)
```

Comment:
I can do a WriteInt32 followed by WriteBlock, but I'd think one of the tests added would fail if that changed. (``Compression.Snappy reads messages that are compatible with reference implementation``)